Foundation/NSRunLoop.swift

```diff
@@ -95,10 +95,6 @@ extension NSRunLoop {
             return false
         }
         let modeArg = mode._cfObject
-        if _CFRunLoopFinished(_cfRunLoop, modeArg) {
-            return false
-        }
-        
```

Comment:
removing this is contrary to how the darwin version is implemented; this means that there is something seriously incorrect with how the backing for CFRunLoop is functioning if this "fixes the problem".